IBX 9819: Collaborative editing Product Guide added in Developer Documentation (#2715)
ibexa/documentation-developer: docs/content_management/collaborative_editing/collaborative_editing_guide.md

---
description: The Collaborative editing product guide provides a full description of the features and benefits that this module brings to the clients.
---

# Collaborative editing product guide

## What is collaborative editing

Collaborative editing is a feature that enables multiple users to work on the same content simultaneously - edit specific parts of the content or review them.
The system automatically tracks changes, allowing seamless collaboration within a single content item.

> Users can edit and review content in real time, making teamwork faster, more efficient, and streamlining the content review process.

Reviewer: I'd add a sentence here about sharing/previewing, this is a nice feature when working with external users (3rd party agency, for example).

## Availability

> Collaborative editing is available in all [[= product_name =]] editions.

Reviewer: We need to confirm this, I think a separate deal might have to be signed to have access to real-time editing in richtext.

Sharing (preview) works out of the box I think?

## Prerequisites

To use the **Copy link** option, which allows you to copy a link to the clipboard and share it through communication channels with other users, the Clipboard API is required.
As a result, this option may not work in some browsers, such as Safari.

## How does collaborative editing work

> Collaborative editing allows to work together on the same content item.

Reviewer: ```suggestion
Collaborative editing allows to work together on the same content items or products.
```

This is done through a collaboration session.

> When you create a new draft of content item you can invite other users to join a collaboration session, thanks to [CKEditor collaboration features](https://ckeditor.com/ckeditor-5/capabilities/collaboration-features/).

Reviewer: ```suggestion
When you create a new draft of a content item you can invite other users to join a collaboration session, thanks to [CKEditor collaboration features](https://ckeditor.com/ckeditor-5/capabilities/collaboration-features/).
```

This action generates a unique session for that draft.

You can invite other users to join the session, both internal and external:

- **Internal** - by searching their name or email address. These users can either edit the content item or preview it, depending on your choice.
- **External** - by providing their email address in the field. They can only preview the content item.

Once they accept the invitation, they are able to join you in editing content item or reviewing it.

> ![Collaborative editing - invitation](img/collaboration_invitation.png)

Reviewer: Image is missing I think? Will be able to check this once the preview works.

You can change the users access or remove it at any time.

Additionally, you can share a direct link to the collaborative session using the **Copy link** button.
Link is copied to the clipboard and you can share it with the users through communication channels.

After inviting users to a collaboration session, they receive a notification visible on the main dashboard or by email.

> ### Real-time editing

Reviewer: As of now, I see no difference between collaborative editing (defined above) and real-time editing (defined here)

Both mention tracking real-time changes.

> Real-time collaboration is a premium feature that works by syncing changes in real time, so everyone can see updates instantly.

Reviewer: "premium" in which sense? There's no mention of an additional deal that's needed, the feature is described as available for everyone.

Avatars of the users invited to collaboration session are visible at the top of the editing screen, also in distraction free mode.
While editing Rich Text fields, you can see colored tracking tags with user avatar thumbnails that indicate who is currently working on it.

Everyone in the session can see each other's updates as they happen — no more switching between tools or waiting for feedback.
This makes content creation and review faster, more interactive, and much easier to manage as a team.

## Capabilities

> ### Editing content item

Reviewer: ```suggestion
### Editing content items and products
```

> Collaborative editing is enabled in Rich Text fields.

Reviewer: I'd link to Rich Text field type here (or the online editor product guide)

Other fields are disabled and can be only edited by the Owner of the content item.

> Collaboration is available for the following content types containing Rich Text fields:

Reviewer: ```suggestion
Collaboration is available for the following content or product types with Rich Text fields:
```

- Article
- Folder
- Form
- Product category
- Custom content types

> - Product (without variants)

Reviewer: ```suggestion
- Custom product types (without variants)
```
It's the same case as "custom content types"

## How to get started

To start collaborative editing, you need to invite collaborators using the **Share** button.

Collaboration session begins when first invited user accepts the invitation and joins the session.

## Benefits

### Real-time collaboration

Collaborative editing enables multiple users to work on the same content item at the same time.

> Everyone in the session can see changes as they happen, which eliminates the delays in feedback and updates or waiting for someone to finish editing before others can contribute.

Reviewer: ```suggestion
Everyone in the session can see changes as they happen, which shortens the feedback loop and allows multiple people to work in parallel.
```
I'm not a fan of "which eliminates the delays in feedback and update" in this sentence

### Improved efficiency

By allowing simultaneous editing, the content creation and review process becomes significantly faster.
Team members can work together and finalize content in a fraction of the time it would take with a standard workflow.

### Enhanced teamwork

> All the users invitated to the collaboration session can share ideas, make suggestions, and refine each other’s work in a shared environment, creating a stronger sense of team ownership and collaboration.

Reviewer: ```suggestion
All users invited to the collaboration session can share ideas, make suggestions, and refine each other’s work in a shared environment, creating a stronger sense of team ownership and collaboration.
```

### Seamless feedback loop

Users can share and receive feedback instantly within the same editing interface.
This reduces the need to switch between apps or track changes manually.

### Simplified content review process

With real-time access to the same draft, reviewers can jump in, provide input, and approve content more quickly.
This streamlines the entire review cycle and minimizes delays caused by version confusion or slow feedback.

### Cross-functional collaboration

Collaborative editing allows teams to involve users from across the organization.
Their input can be integrated directly into the editing process, leading to more accurate and aligned content.

### Change tracking

All changes made by the editors belonging to the session are automatically tracked, so you can see what was edited, by whom, and when.

> This provides full transparency and makes it easy to review.

Reviewer: ```suggestion
This provides full transparency and makes the review process easier.
```
To avoid "it" - it is be "the changes" in this case?

---
description: The Collaborative editing product guide provides a full description of the features and benefits that this module brings to the clients.
---

# Collaborative editing product guide

## What is collaborative editing

Collaborative editing is a feature that allows multiple users to work on the same content or product - whether to preview, review, or edit it.

> By giving users access to preview the content before it's published, review and collaboration becomes much easier.

Reviewer: ```suggestion
By giving users access to preview the content before it's published, review and collaboration become much easier.
```

When you create a new draft of a content item you can invite other users to join a collaboration session, thanks to [CKEditor collaboration features](https://ckeditor.com/ckeditor-5/capabilities/collaboration-features/).
This action generates a unique session for that draft.
Collaboration session begins when first invited user accepts the invitation and joins the session.

To start collaborative editing, you need to invite collaborators using the **Share** button.

You can invite other users to join the session, both internal and external:

- **Internal** - by searching their name or email address. These users can either edit the content item or preview it, depending on your choice.
- **External** - by providing their email address in the field. They can only preview the content item.

Once they accept the invitation, they are able to join you in editing content item or reviewing it.

![Collaborative editing - invitation](img/collaboration_invitation.png)

You can change the users access or remove it at any time.

After inviting users to a collaboration session, they receive a notification visible on the main dashboard or by email.

### Real-time editing

> Real-time editing is an advanced part of the Collaboration feature available for *five users*.

Reviewer: ```suggestion
Real-time editing is an advanced part of the Collaboration feature, allowing *up to five users* to work together in a single session.
```
I'd clarify the "5 users" part - is it "up 5 users in a single session"/"5 users across all sessions that are active"/"only 5 chosen editors can use it"/"something else"

(I'm not sure the "up to 5 at the same time" is right)
